opentelemetry-sdk/src/opentelemetry/sdk/trace/sampling.py

```diff
@@ -413,14 +486,470 @@ class _ParentBasedAlwaysOn(ParentBased):
+            self.sampler.update(per_operation_strategies)
+        else:
+            self.sampler = AdaptiveSampler(per_operation_strategies, self.max_operations)
+        # TODO(sconover) there's a general question about what to do about metrics...
+        # self.metrics.sampler_updated(1)
+
+    def _update_rate_limiting_or_probabilistic_sampler(self, response):
+        s_type = response.get(STRATEGY_TYPE_STR)
+        new_sampler = self.sampler
+        if s_type == PROBABILISTIC_SAMPLING_STRATEGY:
+            sampling_rate = get_sampling_probability(response)
+            new_sampler = TraceIdRatioBased(rate=sampling_rate)
+        elif s_type == RATE_LIMITING_SAMPLING_STRATEGY:
+            mtps = get_rate_limit(response)
+            if mtps < 0 or mtps >= 500:
+                raise ValueError(
+                    'Rate limiting parameter not in [0, 500) range: %s' % mtps)
+            if isinstance(self.sampler, RateLimitingSampler):
+                if self.sampler.update(max_traces_per_second=mtps):
+                    pass
+                    # TODO(sconover) there's a general question about what to do about metrics...
+                    # self.metrics.sampler_updated(1)
+            else:
+                new_sampler = RateLimitingSampler(max_traces_per_second=mtps)
+        else:
+            raise ValueError('Unsupported sampling strategy type: %s' % s_type)
+
+        if self.sampler != new_sampler:
+            self.sampler = new_sampler
+            # TODO(sconover) there's a general question about what to do about metrics...
+            # self.metrics.sampler_updated(1)
+
+    def _poll_sampling_manager(self):
+        self.logger.debug('Requesting tracing sampler refresh')
+        fut = self._channel.request_sampling_strategy(self.service_name)
+        fut.add_done_callback(self._sampling_request_callback)
+
+    def close(self) -> None:
+        with self.lock:
+            self.running = False
+            if self.periodic:
+                self.periodic.stop()
+
+    def get_description(self) -> str:
+        return f'RateLimitingSampler{{{self.traces_per_second}}}'
+
+    def __str__(self) -> str:
+        return self.get_description()
+    
+def get_sampling_probability(strategy: Optional[Dict[str, Any]] = None) -> float:
+    if not strategy:
+        return DEFAULT_SAMPLING_PROBABILITY
+    probability_strategy = strategy.get(PROBABILISTIC_SAMPLING_STR)
+    if not probability_strategy:
+        return DEFAULT_SAMPLING_PROBABILITY
+    return probability_strategy.get(SAMPLING_RATE_STR, DEFAULT_SAMPLING_PROBABILITY)
+
+def get_rate_limit(strategy: Optional[Dict[str, Any]] = None) -> float:
+    if not strategy:
+        return DEFAULT_LOWER_BOUND
+    rate_limit_strategy = strategy.get(RATE_LIMITING_SAMPLING_STR)
+    if not rate_limit_strategy:
+        return DEFAULT_LOWER_BOUND
+    return rate_limit_strategy.get(MAX_TRACES_PER_SECOND_STR, DEFAULT_LOWER_BOUND)
 
 _KNOWN_SAMPLERS = {
```

Comment:
@srikanthccv This is a comprehensive list of samplers, which is checked in `_get_from_env_or_default`. Obviously this works against the idea of there being samplers from outside of this repo - what do you suggest I do about this?